### PR TITLE
Tested & Improved beans_close_markup and beans_close_markup_e

### PR DESCRIPTION
**Pull request by hellofromtonya**

### CODE REVIEW

File: lib/api/html/functions.php [hunk 1]
```diff
@@ -235,63 +235,64 @@ function beans_selfclose_markup_e( $id, $tag, $attributes = array() ) {
...
  *
  * Note: You can pass additional arguments to the functions that are hooked to <tt>$id</tt>.
  *
  * @since 1.0.0
  *
  * @param string $id  Identical to the opening markup ID.
- * @param string $tag The HTML tag.
+ * @param string $tag The HTML tag. If set to False or empty, the markup HTML tag will be removed but
+ *                    the actions hook will be called. If set the Null, both markup HTML tag and actions
+ *                    hooks will be removed.
```
Comment (christophherr) [inline lib/api/html/functions.php]: Should False and Null be capitalized here?
`If set to False or empty, the HTML tag markup will be removed but the actions hook will be called.
If set to Null, both HTML tag markup and actions hook will be removed.`

File: lib/api/html/functions.php [hunk 2]
```diff
@@ -235,63 +235,64 @@ function beans_selfclose_markup_e( $id, $tag, $attributes = array() ) {
...
- *
- * Note: You can pass additional arguments to the functions that are hooked to <tt>$id</tt>.
+ * Echo the closing HTML tag's markup.  This function is a wrapper for {@see beans_close_markup()}.  See
+ * {@see beans_close_markup()} for more details.
  *
  * @since 1.4.0
  *
  * @param string $id  Identical to the opening markup ID.
- * @param string $tag The HTML tag.
+ * @param string $tag The HTML tag. If set to False or empty, the markup HTML tag will be removed but
```
Comment (christophherr) [inline lib/api/html/functions.php]: See the previous comment.

File: tests/phpunit/integration/api/html/beansCloseMarkup.php [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+	 */
+	public function test_should_return_built_html_when_append_or_after_hooks_not_registered() {
+		$actual = beans_close_markup( 'beans_archive_title', 'h1' );
+		$this->assertSame( '</h1>', $actual );
+		$this->assertEquals( 0, did_action( 'beans_archive_title_append_markup' ) );
+		$this->assertEquals( 0, did_action( 'beans_archive_title_after_markup' ) );
+	}
+
+	/**
+	 * Test beans_close_markup() should return a built HTML when append or after hooks are registered.
```
Comment (christophherr) [inline tests/phpunit/integration/api/html/beansCloseMarkup.php]: `the built HTML`

File: tests/phpunit/integration/api/html/beansCloseMarkupE.php [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+		$actual = ob_get_clean();
+
+		$this->assertSame( '</h1>', $actual );
+		$this->assertEquals( 0, did_action( 'beans_archive_title_append_markup' ) );
+		$this->assertEquals( 0, did_action( 'beans_archive_title_after_markup' ) );
+	}
+
+	/**
+	 * Test beans_close_markup_e() should echo the before, element, and prepend HTML when callbacks are registered
+	 * the "_append_markup" and "_after_markup" hooks.
```
Comment (christophherr) [inline tests/phpunit/integration/api/html/beansCloseMarkupE.php]: `to the...`

File: tests/phpunit/unit/api/html/beansCloseMarkup.php [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+	 */
+	public function test_should_return_built_html_when_append_or_after_hooks_not_registered() {
+		Monkey\Functions\when( 'beans_apply_filters' )->returnArg( 2 );
+		Monkey\Functions\when( '_beans_render_action' )->justReturn( '' );
+
+		$this->assertSame( '</h1>', beans_close_markup( 'beans_archive_title', 'h1' ) );
+	}
+
+	/**
+	 * Test beans_open_markup() should return a built HTML when append or after hooks are registered.
```
Comment (christophherr) [inline tests/phpunit/unit/api/html/beansCloseMarkup.php]: `the built HTML`

File: tests/phpunit/unit/api/html/beansCloseMarkupE.php [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+		Monkey\Functions\when( '_beans_render_action' )->justReturn( '' );
+
+		ob_start();
+		beans_close_markup_e( 'beans_archive_title', 'h1' );
+		$this->assertSame( '</h1>', ob_get_clean() );
+	}
+
+	/**
+	 * Test beans_close_markup_e() should echo the before, element, and prepend HTML when callbacks are registered
+	 * the "_append_markup" and "_after_markup" hooks.
```
Comment (christophherr) [inline tests/phpunit/unit/api/html/beansCloseMarkupE.php]: `to the`

File: tests/phpunit/integration/api/html/beansCloseMarkup.php [hunk 7]
```diff
@@ -99,10 +101,19 @@ public function test_should_return_built_html_when_append_or_after_hooks() {
...
-	 * Test beans_close_markup() should return only the output from the hooked callbacks and not the HTML element when
+	 * Test beans_close_markup() should return an empty string when (1) the tag is an empty string and (2) a callback
+	 * is not registered to either the "_append_markup" or "_after_markup" hook.
+	 */
+	public function test_should_return_empty_string_when_empty_tag_and_callback_not_registered_to_hooks() {
+		$this->assertSame( '', beans_close_markup( 'beans_archive_title', '' ) );
+	}
+
+	/**
+	/**
```
[the remark this answers is not in the document]
Comment (hellofromtonya) [reply tests/phpunit/integration/api/html/beansCloseMarkup.php]: Doh.

**Pull request merged**